src/Interpreters/MergeJoin.cpp

```diff
@@ -23,12 +25,50 @@ namespace ErrorCodes
 {
     extern const int NOT_IMPLEMENTED;
     extern const int PARAMETER_OUT_OF_BOUND;
+    extern const int ILLEGAL_COLUMN;
     extern const int LOGICAL_ERROR;
 }
 
 namespace
 {
 
+String deriveTempName(const String & name)
+{
+    return "--" + name;
```

Comment:
Why such name?

Reply:
Some internal table name needed, I haven't found good solution for it. This column added at the beginning of function joinBlock and removed at the end, so it's not visible outside. But we haven't some mechanism for generating temporary names that also cannot be already presented in block.
Do you have any ideas how to solve it?